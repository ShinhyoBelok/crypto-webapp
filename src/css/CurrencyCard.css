.currencyCard {
  padding: 0.5rem;
  flex-direction: column;
  width: 50%;
  gap: 2rem;
}
.currencyCard img {
  height: 1.5rem;
  width: 1.5rem;
  filter: brightness(0) invert(1);
}
.currencyCard .top {
  align-items: center;
  justify-content: space-between;
}
.currencyCard .bt {
  flex-direction: column;
  text-align: right;
}
.currencyCard .bt h3 {
  font-size: 1rem;
  font-weight: 300;
}
.currencyCard .rank {
  font-size: 1rem;
  font-weight: 400;
}

.bgOne {
  background-color: #cf4278;
}

.bgTwo {
  background-color: #dc4782;
}