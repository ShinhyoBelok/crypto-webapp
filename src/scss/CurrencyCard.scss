.currencyCard {
  padding: 0.5rem;
  flex-direction: column;
  width: 50%;
  gap: 2rem;

  img {
    height: 1.5rem;
    width: 1.5rem;
    filter: brightness(0) invert(1);
  }

  .top {
    align-items: center;
    justify-content: space-between;
  }

  .bt {
    flex-direction: column;
    text-align: right;

    h3 {
      font-size: 1rem;
      font-weight: 300;
    }
  }

  .rank {
    font-size: 1rem;
    font-weight: 400;
  }
}

.bgOne {
  background-color: #cf4278;
}

.bgTwo {
  background-color: #dc4782;
}
