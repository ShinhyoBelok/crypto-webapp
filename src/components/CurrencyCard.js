import React from 'react';
import { string } from 'prop-types';
import '../css/CurrencyCard.css';

let count = 0;
export default function CurrencyCard(props) {
  const {
    id, name, rank, priceUsd,
  } = props;
  const price = parseFloat(priceUsd, 10).toFixed(5);
  let bgColor;
  if (count === 0) {
    bgColor = 'bgOne';
    count += 1;
  } else if (count === 1) {
    bgColor = 'bgTwo';
    count += 1;
  } else if (count === 2) {
    bgColor = 'bgTwo';
    count += 1;
  } else if (count === 3) {
    bgColor = 'bgOne';
    count = 0;
  }

  return (
    <article className={`currencyCard flex ${bgColor}`} id={id}>
      <div className="top flex">
        <h2 className="rank">
          Rank&ensp;
          {rank}
        </h2>
        <img src="https://cdn-icons-png.flaticon.com/512/1549/1549454.png" alt="flecha" />
      </div>
      <div className="bt flex">
        <h2>{name}</h2>
        <h3>{`USD ${price}`}</h3>
      </div>
    </article>
  );
}

CurrencyCard.propTypes = {
  name: string.isRequired,
  rank: string.isRequired,
  id: string.isRequired,
  priceUsd: string.isRequired,
};
