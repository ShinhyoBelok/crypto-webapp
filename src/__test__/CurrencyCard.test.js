import React from 'react';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import store from '../redux/configureStore';
import CurrencyCard from '../components/CurrencyCard';

describe('CurrencyCard component', () => {
  test('Test CurrencyCard button render correctly', () => {
    const currencyCard = render(
      <Provider store={store}>
        <BrowserRouter>
          <CurrencyCard />
        </BrowserRouter>
      </Provider>,
    );
    expect(currencyCard).toMatchSnapshot();
  });
});