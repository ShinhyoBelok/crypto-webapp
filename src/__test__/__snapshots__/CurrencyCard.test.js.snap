// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CurrencyCard component Test CurrencyCard button render correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <article
        class="currencyCard flex bgOne"
        id="id"
        role="presentation"
      >
        <div
          class="top flex"
        >
          <h2
            class="rank"
          >
            Rank 
            rank
          </h2>
          <img
            alt="flecha"
            src="https://cdn-icons-png.flaticon.com/512/1549/1549454.png"
          />
        </div>
        <div
          class="bt flex"
        >
          <h2>
            name
          </h2>
          <h3>
            USD NaN
          </h3>
        </div>
      </article>
    </div>
  </body>,
  "container": <div>
    <article
      class="currencyCard flex bgOne"
      id="id"
      role="presentation"
    >
      <div
        class="top flex"
      >
        <h2
          class="rank"
        >
          Rank 
          rank
        </h2>
        <img
          alt="flecha"
          src="https://cdn-icons-png.flaticon.com/512/1549/1549454.png"
        />
      </div>
      <div
        class="bt flex"
      >
        <h2>
          name
        </h2>
        <h3>
          USD NaN
        </h3>
      </div>
    </article>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
